operators/multiclusterobservability/controllers/placementrule/manifestwork.go

```diff
@@ -466,114 +467,98 @@ func createUpdateResourcesForHubMetricsCollection(c client.Client, manifests []w
 		log.Info("MCO Operator is terminating, skip creating resources for hub metrics collection")
 		return nil
 	}
+
 	hubManifestCopy = make([]workv1.Manifest, len(manifests))
 	for i, manifest := range manifests {
 		obj := manifest.RawExtension.Object.DeepCopyObject()
 		hubManifestCopy[i] = workv1.Manifest{RawExtension: runtime.RawExtension{Object: obj}}
 		hubManifestCopy[i] = workv1.Manifest{RawExtension: runtime.RawExtension{Object: obj}}
 	}
+
 	for _, manifest := range hubManifestCopy {
 		obj := manifest.RawExtension.Object.(client.Object)
-		if obj.GetObjectKind().GroupVersionKind().Kind == "Namespace" || obj.GetObjectKind().GroupVersionKind().Kind == "ObservabilityAddon" {
-			// We do not want to create ObservabilityAddon and namespace open-cluster-management-add-on observability for hub cluster
+		var currentObj client.Object
+
+		gvk := obj.GetObjectKind().GroupVersionKind()
+		switch gvk.Kind {
+		case "Namespace", "ObservabilityAddon":
+			// Skip these kinds
 			continue
-		}
-		kind := obj.GetObjectKind().GroupVersionKind().Kind
-		if kind != "ClusterRole" && kind != "ClusterRoleBinding" && kind != "CustomResourceDefinition" {
+		case "ClusterRole", "ClusterRoleBinding", "CustomResourceDefinition":
+			// No namespace needed for these kinds
+		default:
+			// Set default namespace for other kinds
 			obj.SetNamespace(config.GetDefaultNamespace())
 		}
-		if obj.GetObjectKind().GroupVersionKind().Kind == "ClusterRoleBinding" {
+
+		if gvk.Kind == "ClusterRoleBinding" {
 			role := obj.(*rbacv1.ClusterRoleBinding)
-			role.Subjects[0].Namespace = config.GetDefaultNamespace()
+			if len(role.Subjects) > 0 {
+				role.Subjects[0].Namespace = config.GetDefaultNamespace()
+			}
+		}
+
+		switch obj.GetObjectKind().GroupVersionKind().Kind {
```

Comment:
I would split the code inside the loop into two different loops. At the beginning we filter on some kinds, now on some others. The code is hard to follow.